force_bdss/mco/parameters/mco_parameters.py

```diff
@@ -0,0 +1,147 @@
+import numpy as np
+
+from traits.api import List, Unicode, Property, Float, ReadOnly, Int, Either
+from traitsui.api import ListEditor
+
+from .base_mco_parameter import BaseMCOParameter
+from .base_mco_parameter_factory import BaseMCOParameterFactory
+
+
+class FixedMCOParameter(BaseMCOParameter):
+    """ Fixed MCO parameter for (dummy) constant-valued data. The value
+    must be specified before use: the value is <undefined> by default."""
+
+    #: Fixed parameter value
+    value = ReadOnly
+
+    sample_values = Property(depends_on="value", visible=False)
+
+    def _get_sample_values(self):
+        return [self.value]
+
+
+class FixedMCOParameterFactory(BaseMCOParameterFactory):
+    """ Fixed Parameter factory"""
+
+    #: This identifier must be unique for your parameter.
+    #: Once again you are fully responsible for its uniqueness within the scope
+    #: of the MCO it belongs to. You can have the same identifier if and only
+    #: if they belong to different MCOs.
+    #: Again, you are free to choose a uuid if you so prefer.
+    def get_identifier(self):
+        return "fixed"
+
+    #: A name that will appear in the UI to identify this parameter.
+    def get_name(self):
+        return "Fixed"
+
+    #: Definition of the associated model class.
+    def get_model_class(self):
+        return FixedMCOParameter
+
+    #: A long description of the parameter
+    def get_description(self):
+        return "A parameter with a fixed ReadOnly value."
+
+
+class RangedMCOParameter(BaseMCOParameter):
+    """ Numerical MCO parameter that continuously ranges between
+    lower and upper floating point values."""
+
+    #: Lower bound for parameter values range
+    lower_bound = Float(0.1)
+
+    #: Upper bound for parameter values range
+    upper_bound = Float(100.0)
+
+    n_samples = Int(5)
+
+    sample_values = Property(
+        depends_on="lower_bound,upper_bound,n_samples", visible=False
+    )
+
+    def _get_sample_values(self):
+        return np.linspace(self.lower_bound, self.upper_bound, self.n_samples)
```

Comment:
Can we return this as a list? It's inefficient but at the moment the BaseMCO classes are all working with interfaces involving basic python types.

Reply:
Thank you, this will be `list(np.linspace(...))` instead.